cmd/k8s-operator/ingress.go

```diff
@@ -267,5 +279,28 @@ func (a *IngressReconciler) maybeProvision(ctx context.Context, logger *zap.Suga
 func (a *IngressReconciler) shouldExpose(ing *networkingv1.Ingress) bool {
 	return ing != nil &&
 		ing.Spec.IngressClassName != nil &&
-		*ing.Spec.IngressClassName == "tailscale"
+		*ing.Spec.IngressClassName == tailscaleIngressClassName
+}
+
+// validateIngressClass attempts to validate that 'tailscale' IngressClass
+// included in Tailscale installation manifests exists and has not been modified
+// to attempt to enable features that we do not support.
+func (a *IngressReconciler) validateIngressClass(ctx context.Context) error {
+	ic := &networkingv1.IngressClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: tailscaleIngressClassName,
+		},
+	}
+	if err := a.Get(ctx, client.ObjectKeyFromObject(ic), ic); apierrors.IsNotFound(err) {
```

Comment:
Doing this will start a watch for `IngressClass` as that's what c/r does by default. I don't really see that as an issue since those are small resources of which there shouldn't be many. I do want to clean up our caching logic at some point though, we could cache less and cache metadata only in some cases, but so far no-one has raised memory consumption issues or slowness

Reply:
Should this be a code comment to make sure folks reading this code have more context? It's going to be more visible than a comment on the PR.

Reply:
For this particular one I would rather not leave a comment there it's not really relevant to the code flow. I will add a TODO to re-evaluate caching in where we set up controllers.